src/games/strategy/triplea/ui/RouteOptimizer.java

```diff
@@ -22,60 +22,115 @@
+      // If the Map is not infinite scrolling, we can safely return the given Points
+      endPoint = route[route.length - 1];
+      return route;
+    }
+    List<Point> result = new ArrayList<>();
+    Point previousPoint = null;
+    for (Point point : route) {
+      if (previousPoint == null) {
+        previousPoint = point;
+        result.add(point);
+        continue;
       }
+      Point closestPoint = getClosestPoint(previousPoint, getPossiblePoints(point));
+      result.add(closestPoint);
+      previousPoint = closestPoint;
     }
-    return points;
-  }
-
-  private boolean crossMapSeam(Point... route) {
-    final int minX = minX(route);
-    return Arrays.stream(route).filter(point -> (point.x - minX) > mapPanel.getWidth()).findAny().isPresent();
+    endPoint = result.get(result.size() - 1);
+    return result.toArray(new Point[result.size()]);
   }
 
-  private static int minX(Point... route) {
-    int minX = Integer.MAX_VALUE;
-    for (Point p : Arrays.asList(route)) {
-      if (p.x < minX) {
-        minX = p.x;
+  /**
+   * Returns the Closest Point out of the given Pool
+   * 
+   * @param source the reference Point
+   * @param pool Point2D List with all possible options
+   * @return the closest point in the Pool to the source
+   */
+  private Point getClosestPoint(Point source, List<Point2D> pool) {
+    double closestDistance = Double.MAX_VALUE;
+    Point closestPoint = null;
+    for (Point2D possibleClosestPoint : pool) {
+      if (closestPoint == null) {
+        closestDistance = source.distance(possibleClosestPoint);
+        closestPoint = normalizePoint(getPoint(possibleClosestPoint));
+      } else {
+        double distance = source.distance(possibleClosestPoint);
+        if (closestDistance > distance) {
+          closestPoint = getPoint(possibleClosestPoint);
+          closestDistance = distance;
+        }
       }
     }
-    return minX;
+    return closestPoint;
   }
 
   private Point normalizePoint(Point point) {
     return new Point(point.x % mapWidth, point.y % mapHeight);
   }
 
   /**
+   * Method for getting Points, which are a mapHeight/Width away from the actual Point
+   * Used to display routes with higher offsets than the map width/height
+   * 
+   * @param point The Point to "clone"
+   * @return A List of all possible Points depending in map Properties
+   *         size may vary
+   */
+  private List<Point2D> getPossiblePoints(Point2D point) {
+    List<Point2D> result = new ArrayList<>();
+    result.add(point);
+    if (isInfiniteX && isInfiniteY) {
+      result.addAll(Arrays.asList(
+          new Point2D.Double(point.getX() - mapWidth, point.getY() - mapHeight),
+          new Point2D.Double(point.getX() - mapWidth, point.getY() + mapHeight),
+          new Point2D.Double(point.getX() + mapWidth, point.getY() - mapHeight),
+          new Point2D.Double(point.getX() + mapWidth, point.getY() + mapHeight)));
+    }
+    if (isInfiniteX) {
```

Comment:
Okay, please note that in the code as a comment so we have that context there. (I prefer when code is self documenting through method and variable names, but the point is to get the contextual information of why this is being done captured in the code one way or another.) I recently had to look at a git history, back track changes to their commit dates, look for PRs on those dates to then find review discussion that actually told me why some code was the way it was.  You can probably guess that was an enjoyable and efficient process ;)